src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -35,7 +36,25 @@ public class AddressBook implements ReadOnlyAddressBook {
         transactions = new UniqueTransactionList();
     }
 
-    public AddressBook() {}
+    private final Comparator<Person> desc = (p1, p2) -> {
+        int balCompare = -transactions.getBalance(p1.getName()).compareTo(transactions.getBalance(p2.getName()));
+        if (balCompare == 0) {
+            return p1.getName().compareTo(p2.getName());
+        }
+        return balCompare;
+    };
+    private final Comparator<Person> asc = (p1, p2) -> {
+        int balCompare = transactions.getBalance(p1.getName()).compareTo(transactions.getBalance(p2.getName()));
+        if (balCompare == 0) {
+            return p1.getName().compareTo(p2.getName());
+        }
+        return balCompare;
+    };
```

Comment:
Should these be defined in the `setDesc` and `setAsc` functions instead of being defined as instance fields?

Reply:
Done